### PR TITLE
test(pkg/regionspan): migrate test-infra to testify for pkg/regionspan

### LINKED ISSUE #2927: migrate test-infra to testify for `pkg/regionspan` pkg

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve? 
migrate test-infra to testify for pkg/regionspan,  close  #2927

### What is changed and how it works?

1. migrate test-infra from gocheck to testify
2. migrate leak-test from testleak to goleak

### Check List 

Tests 
 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by maxshuang** [comment]
@3AceShowHand @CharlesCheung96 @amyangfei    PLAL. Thanks

File: pkg/regionspan/span_test.go [hunk 1]
```diff
@@ -14,21 +14,17 @@
 package regionspan
 
 import (
+	"bytes"
 	"testing"
 
-	"github.com/pingcap/check"
-	"github.com/pingcap/ticdc/pkg/util/testleak"
+	"github.com/stretchr/testify/require"
+
```
Comment (amyangfei) [inline pkg/regionspan/span_test.go:21]: please remove the blank line

File: pkg/regionspan/region_range_lock_test.go [hunk 2]
```diff
@@ -16,47 +16,43 @@ package regionspan
...
+func mustSuccess(t *testing.T, res LockRangeResult, expectedCheckpointTs uint64) {
+	require.Equal(t, res.Status, LockRangeStatusSuccess)
+	require.Equal(t, res.CheckpointTs, expectedCheckpointTs)
 }
 
-func mustStale(c *check.C, res LockRangeResult, expectedRetryRanges ...ComparableSpan) {
-	c.Assert(res.Status, check.Equals, LockRangeStatusStale)
-	c.Assert(res.RetryRanges, check.DeepEquals, expectedRetryRanges)
+func mustStale(t *testing.T, res LockRangeResult, expectedRetryRanges ...ComparableSpan) {
+	require.Equal(t, res.Status, LockRangeStatusStale)
```
[the remark this answers is not in the document]
Comment (maxshuang) [reply pkg/regionspan/region_range_lock_test.go:31]: Sure

[the remark this answers is not in the document]
Comment (maxshuang) [reply pkg/regionspan/region_range_lock_test.go]: done.  Please help check.

**Review by asddongmen — APPROVED** [review]

**Pull request merged**